### ISSUE TITLE
Problem. Missing API request to get tweet html (twitter)

### ISSUE THREAD

**Issue by georgeciubotaru** [holdex/holdex-venture-studio#313]
Scope:

- Create API endpoint `/api/twitter/tweet-lookup`
- Using Twitter API get lookup tweet json
- Implement parser similar to: https://github.com/holdex/twitter-widget/blob/main/src/components/lib/fetcher.js

Docs: https://developer.twitter.com/en/docs/twitter-api/tweets/lookup/quick-start

**Comment by TerrenceAddison** [comment]
@georgeciubotaru just would like to ask clarification for this task.  API Endpoint is already ready is that correct?
 and so under the scope I just need to do step 2 and step 3?

**Comment by georgeciubotaru** [comment]
@TerrenceAddison Yes, the file with the API is ready, you just need to do steps 2 and 3, and also, update the file from step 1

**Comment by TerrenceAddison** [comment]
@georgeciubotaru alright give me another tag if suraj's PR is merged, will ask further questions if needed

**Comment by georgeciubotaru** [comment]
@TerrenceAddison PR merged

**Comment by surajsharma** [comment]
@georgeciubotaru this can be closed

**Comment by georgeciubotaru** [comment]
@surajsharma No, Terrence is working in this task

**Comment by TerrenceAddison** [comment]
@surajsharma can I ask for the URL where I can see the site that uses the twitter API?
@georgeciubotaru since we want to use the twitter API, do you already have a set of keys for me to use? or do I sign up for a dev twitter acc myself?

**Comment by surajsharma** [comment]
@TerrenceAddison you can use the following staging link for this purpose https://holdex-venture-studio-2x6onp1ss-holdex-accelerator.vercel.app/c/elysio/testing-multiple-tweets

**Comment by georgeciubotaru** [comment]
@TerrenceAddison

https://github.com/holdex/holdex-venture-studio/issues/60#issuecomment-1561043917

Regarding the API key, re-run the server and the new env will be available (be careful, the variable should be declared in the `lib/server/config` file and used only in server components)

**Comment by TerrenceAddison** [comment]
> @TerrenceAddison
> 
> [#60 (comment)](https://github.com/holdex/holdex-venture-studio/issues/60#issuecomment-1561043917)
> 
> Regarding the API key, re-run the server and the new env will be available (be careful, the variable should be declared in the `lib/server/config` file and used only in server components)

Can I ask for clarification which is the twitter dev key on the `.env` file? I don't see any that collerates with `twitter`

**Comment by TerrenceAddison** [comment]
@georgeciubotaru would like to clarify the task that I would need to do to complete this.
1. create new `tweet-lookup.server.ts` under `api/tweets/` folder to make use the twitter tweet-lookup API
2. implement parser for the data of from the api response based on the above link given (I assume it would be better to be placed in the utils folder?)

Am I missing/wrong about anything

**Comment by georgeciubotaru** [comment]
@TerrenceAddison All good
1. use the existing file `api/tweets`
2. Implement Twitter API -> get a response and parse it -> return the parsed payload

**Comment by georgeciubotaru** [comment]
@TerrenceAddison Please leave an estimation for this task. (if possible)

**Comment by TerrenceAddison** [comment]
> @TerrenceAddison Please leave an estimation for this task. (if possible)

I'm currently a bit busy so I'll try to get it done over the weekend

> @TerrenceAddison All good
> 
> 1. use the existing file `api/tweets`
> 2. Implement Twitter API -> get a response and parse it -> return the parsed payload

so i'll modify the api/tweets/+server.ts file to use the twitter API is that correct?

**Comment by georgeciubotaru** [comment]
@TerrenceAddison Yes

**Comment by TerrenceAddison** [comment]
@georgeciubotaru Yesterday the bearer token was working while testing from postman but now it's giving this error, can you check and create a project / app from the dev acc?
```
{
    "client_id": "24094107",
    "detail": "When authenticating requests to the Twitter API v2 endpoints, you must use keys and tokens from a Twitter developer App that is attached to a Project. You can create a project via the developer portal.",
    "registration_url": "https://developer.twitter.com/en/docs/projects/overview",
    "title": "Client Forbidden",
    "required_enrollment": "Appropriate Level of API Access",
    "reason": "client-not-enrolled",
    "type": "https://api.twitter.com/2/problems/client-forbidden"
}
```

**Comment by TerrenceAddison** [comment]
@georgeciubotaru just following up if you did any checks? cause I can't continue without the key working

**Comment by georgeciubotaru** [comment]
@TerrenceAddison Terrence, you are right, Twitter has changed the access level for `Free` accounts, and we don't have access to the `Twitter lookup` request.

Let's use this fallback URL: https://developer.twitter.com/en/docs/twitter-api/v1/tweets/post-and-engage/api-reference/get-statuses-lookup

**Comment by TerrenceAddison** [comment]
@georgeciubotaru I was reading the fetcher.js and there exists the line
`const { data, includes } = tweetJSON;` I was running it and it seems `includes` is not an available value from the twitter API. Wonder if you can give me some insight on the usage?

**Comment by georgeciubotaru** [comment]
@TerrenceAddison How about to check the docs?
https://developer.twitter.com/en/docs/twitter-api/migrate/data-formats/standard-v1-1-to-v2

**Comment by TerrenceAddison** [comment]
@georgeciubotaru  to clarify was the widget using v2, so now I also have to translate to v1.1 also?

**Comment by georgeciubotaru** [comment]
Log: The task is free to take.

**Comment by predrag-us** [comment]
Moving to `Ready` since the issue is untaken.

**Comment by zolotokrylin** [comment]
@georgeciubotaru Twitter API widget is dead. See https://github.com/holdex/holdex-venture-studio/issues/434

**Comment by zolotokrylin** [comment]
@georgeciubotaru can we resolved this issue this week?

**Comment by georgeciubotaru** [comment]
@zolotokrylin 

The request we need from Twitter is only available in (v2) for Basic Plan (100$ per month)
V1 seems to be deprecated (I'm unable to test because cannot get a v1 key anymore)


solution: Hide the Twitter widget from `BodyRenderer` and replace it with a simple link component.

**Comment by georgeciubotaru** [comment]
@zolotokrylin Let me know if the solution is suitable for us, and feel free to create a problem if any issues.

**Comment by zolotokrylin** [comment]
@georgeciubotaru why we can't use API v1? I am checking documentation and it looks like there is a tweet look up endpoint.

**Comment by georgeciubotaru** [comment]
@zolotokrylin Tweets lookup is V2. V1 is partially deprecated, and cannot check it because of the missing API key for V1, which cannot be obtained anymore.

2 solutions:

- Buy a Basic Pack for V2
- https://developer.twitter.com/en/docs/twitter-for-websites/embedded-tweets/overview
